.github/workflows/dispatch-yocto-build-deploy.yml

```diff
@@ -0,0 +1,124 @@
+name: "Test Yocto"
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+    branches:
+      - "main"
+      - "master"
+  pull_request_target:
+    types: [opened, synchronize]
+    branches:
+      - "main"
+      - "master"
+
+  workflow_dispatch:
+    # you may only define up to 10 `inputs` for a `workflow_dispatch` event
+    inputs:
+      device-repo:
+        description: balenaOS device repository (owner/repo)
+        required: true
+        type: string
+        default: balena-os/balena-generic
+      device-repo-ref:
+        description: balenaOS device repository tag, branch, or commit to build
+        required: false
+        type: string
+        default: master
+      meta-balena-ref:
+        description: meta-balena ref if not the currently pinned version
+        required: false
+        type: string
+      yocto-scripts-ref:
+        description: balena-yocto-scripts ref if not the currently pinned version
+        required: false
+        type: string
+      machine:
+        description: yocto board name
+        required: true
+        type: string
+        default: generic-amd64
+      environment:
```

Comment:
We need to take into account how this will change once deploys work over the API.

Reply:
We still want to support both production and staging, with different keys and paths for both. The current plan is to use GitHub Environments to set these at the repo/org level and not need to provide them via these workflows at all.

See https://github.com/balena-os/.github/pull/8